### ISSUE TITLE
Getting no GC metrics on .NET5.0

### ISSUE THREAD

**Issue by stefannikolei** [elastic/apm-agent-dotnet#1195]
Using .NET5.0 and Elastic.Apm.NetCoreAll v1.8.0

I used created a new WebApplication from Visual Studio and just added app.UseAllElasticApm(configuration).

I am not getting any gc metrics.
The metric logs look like this:

```
{MetricsCollector} Metrics collected: MetricSample{system.process.cpu.total.norm.pct: 0,0005209173955098381}, MetricSample{system.cpu.total.norm.pct: 0,16761072158813475}, MetricSample{system.process.memory.size: 2225303339008}, MetricSample{system.process.memory.rss.bytes: 99045376}, MetricSample{system.memory.actual.free: 10745376768}, MetricSample{system.memory.total: 34278375424}
```
Configuration looks like this 
```
"ElasticApm":{
"active":true,
"LogLevel": Debug,
"ServerUrl": xxx,
"SecretToken": xxx,
"Environment": "dev",
"ServiceName": "GcTest",
"VerifyServerCert": "false",
"TransactionSampleRate": 1.0
}
```

Do i need to configure something else to get gc information

**Comment by stefannikolei** [comment]
I only get the entry of the GcEventListener that it was initialized. After this i never see anything from him again (LogLevel: trace)

**Comment by gregkalapos** [comment]
I merged #1196 and opened https://github.com/elastic/apm-agent-dotnet/pull/1199 to make sure this is also covered by our tests. 

I'll keep this open until https://github.com/elastic/apm-agent-dotnet/pull/1199 is merged.

**Comment by gregkalapos** [comment]
Test which covers this is merged now in #1199. Thanks for opening this @stefannikolei  and also thanks for the fix.
